force-app/main/default/classes/CRM_JournalUtilities.cls

```diff
@@ -42,65 +42,49 @@ public class CRM_JournalUtilities {
+        if (httpResp.getStatusCode() == 200 || httpResp.getStatusCode() == 201 || httpResp.getStatusCode() == 409) {
             Map<String, Object> responseMap = (Map<String, Object>) JSON.deserializeUntyped(httpResp.getBody());
+            resp.success = true;
             resp.journalpostId = (String) responseMap.get('journalpostId');
-            Boolean journalCompleted = (Boolean) responseMap.get('journalpostferdigstilt');
-            resp.journalPostStatus = journalCompleted ? 'Completed' : 'In Progress';
-        } else if (statusCode == 409) {
+            resp.journalPostStatus = ((Boolean) responseMap.get('journalpostferdigstilt'))
+                ? 'Completed'
+                : 'In Progress';
+
             // Status 409 is only used when the externalReference we provided for the request has already been used
-            Map<String, Object> responseMap = (Map<String, Object>) JSON.deserializeUntyped(httpResp.getBody());
-            String journalExtId = (String) responseMap.get('journalpostId');
-            List<Journal_entry__c> existingJEntries = [
-                SELECT Id
-                FROM Journal_entry__c
-                WHERE Journal_entry_id__c = :journalExtId
-                LIMIT 1
-            ];
-            if (existingJEntries.size() > 0) {
-                resp.success = true;
-                resp.duplicate = existingJEntries[0].Id != referenceId;
-                Boolean journalCompleted = (Boolean) responseMap.get('journalpostferdigstilt');
-                resp.journalPostStatus = journalCompleted ? 'Completed' : 'In Progress';
-            } else {
-                resp.success = false;
-                LoggerUtility logger = new LoggerUtility('Journal');
-                SObject objectRef = String.isNotBlank(referenceId)
-                    ? referenceId.getSobjectType().newSObject(referenceId)
-                    : null;
-
-                logger.httpError(
-                    'Error posting journal entry, got conflict but found no existing journal post',
-                    httpResp,
-                    objectRef,
-                    CRM_ApplicationDomain.Domain.NKS
-                );
-                String errorRef = logger.peek().UUID__c; //Unique error reference for troubleshooting
-                logger.publish();
-                resp.errorMessage = errorRef;
+            if (httpResp.getStatusCode() == 409) {
+                List<Journal_Entry__c> existingJEntries = [
+                    SELECT CRM_Thread__c, CRM_Conversation_Note__c
+                    FROM Journal_Entry__c
+                    WHERE Journal_Entry_Id__c = :resp.journalpostId
+                    LIMIT 1
+                ];
+                resp.duplicate =
+                    existingJEntries.size() > 0 &&
+                    (existingJEntries[0].CRM_Thread__c == referenceId ||
+                    existingJEntries[0].CRM_Conversation_Note__c == referenceId);
```

Comment:
Det blir feil å sjekke Thread og Conversation Note her. Det er ganske spesifikk NKS logikk, som ikke skal påvirke den generelle Journalføringen for andre team

Reply:
Da bør vi i så fall fjerne duplikat-logikk i Journal Conersation Note og Journal Conversation Note v2 flowene, ettersom vi ikke har noe å sjekke mot der.